packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1711,7 +1711,14 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       return;
     }
 
-    _replaceText(ReplaceTextIntent(textEditingValue, data.text!, selection, cause));
+    // After the paste, the cursor should be collapsed and located after the
+    // pasted content.
+    final int lastSelectionIndex = math.max(selection.baseOffset, selection.extentOffset);
+    final TextEditingValue collapsedTextEditingValue = textEditingValue.copyWith(
+      selection: TextSelection.collapsed(offset: lastSelectionIndex),
+    );
+
+    _replaceText(ReplaceTextIntent(collapsedTextEditingValue, data.text!, selection, cause));
```

Comment:
the first arg of the `ReplaceTextIntent` constructor should be the current `TextEditingValue` iirc. I feel it would easier to understand if we just call `userUpdateTextEditingValue` here with the updated text editing value.

Reply:
Agreed, that feels more in the spirit of this stuff.  Will do.